cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/table/field/FieldType.java

```diff
@@ -32,12 +33,22 @@
    * Supported data types.
    */
   public enum Type {
-    INTEGER,
-    LONG,
-    FLOAT,
-    DOUBLE,
-    STRING,
-    BYTES
+    INTEGER(Collections.emptySet()),
+    LONG(Collections.singleton(INTEGER)),
+    FLOAT(Collections.emptySet()),
+    DOUBLE(Collections.singleton(FLOAT)),
+    STRING(Collections.emptySet()),
+    BYTES(Collections.emptySet());
+
+    private final Set<Type> compatibleTypes;
+
+    Type(Set<Type> compatibleTypes) {
```

Comment:
I am curious about why this change is preferred. There might be some databases which forces on the actual type, i.e, not allow writing int type to long(though I haven't heard any). Will this change prevent the extension to be working?